go/extra/stats/cmd/stats.go

```diff
@@ -248,74 +282,84 @@ func getStats(ctx context.Context, consensus consensusAPI.ClientBackend, registr
 		}
 
 		// Get block.
-		block, err := consensus.GetBlock(ctx, height)
-		if err != nil {
-			logger.Error("failed to query block",
-				"err", err,
-				"height", height,
-			)
-			os.Exit(1)
-		}
-		var tmBlockMeta tmApi.BlockMeta
-		if err := cbor.Unmarshal(block.Meta, &tmBlockMeta); err != nil {
-			logger.Error("unmarshal error",
-				"meta", block.Meta,
-				"err", err,
-			)
-			os.Exit(1)
-		}
-
-		// Go over all signatures for a block.
-		for _, sig := range tmBlockMeta.LastCommit.Signatures {
-			if sig.Absent() {
-				logger.Debug("skipping absent signature")
-				continue
-			}
-			nodeTmAddr := sig.ValidatorAddress.String()
-
-			// Signatures are for previous height.
-			if err := ensureNodeTracking(ctx, stats, nodeTmAddr, height-1, registry); err != nil {
-				logger.Error("failed to query registry",
+		tmBlockMeta := getTmBlockMetaOrExit(ctx, consensus, height)
+
+		// Process the commit that is put on chain in this block.
+		if height > 1 {
+			// Commit is for previous height.
+			lastCommitHeight := height - 1
+
+			// Get validators.
+			// Hypothesis: this gets the validator set with priorities after they're incremented to get the round 0
```

Comment:
I assume you checked that this is actually the case?

Reply:
I get the gist of it now